react/Image.tsx

```diff
@@ -114,20 +122,46 @@ function Image(props: ImageProps) {
    */
   const shouldOpenLinkInNewTab = link?.newTab ?? link?.openNewTab
 
+  const { push } = usePixel()
+
+  const imageInfo: any = {
+    src: formattedSrc,
+    alt: formattedAlt,
+    maxWidth,
+    maxHeight,
+    minWidth,
+    minHeight,
+    width,
+    height,
+    srcSet,
+    sizes,
+    link,
+    title,
+    position,
+    ...otherProps
+  }
   const maybeLink = link ? (
     <a
       href={formatIOMessage({ id: link.url, intl })}
       rel={link.noFollow ? 'nofollow' : ''}
       target={shouldOpenLinkInNewTab ? '_blank' : undefined}
       title={formatIOMessage({ id: link?.attributeTitle, intl })}
       className={handles.imageElementLink}
+      onClick={() => customPixelClickEventId && push({ event: customPixelClickEventId, imageInfo})}
     >
       {imgElement}
     </a>
   ) : (
     <Fragment>{imgElement}</Fragment>
   )
 
+  useEffect(() => {
+    customPixelShowEventId && push({
```

Comment:
This should only be called if the image comes up in the viewport 

We have a hook that we can use it:
https://github.com/vtex-apps/search-result/blob/master/react/hooks/useOnView.ts

Example of usage:
https://github.com/vtex-apps/search-result/blob/9b3f0766cec835c1709328fcfd581383dd9382b6/react/hooks/useRenderOnView.tsx#L13-L19

Reply:
Thanks, I changed the code in order to use that hook and it works smoothly. I have 2 things to say about it tho':
 First I had to add // eslint-disable-next-line @typescript-eslint/no-use-before-define in order to work (and in the other projects it does not exists, maybe because that project has an older eslint) the reason is that the eslint does not recognise the good pattern that the refs of the function are used inside another function so in fact there is not "use before declaration" as the their documentation also explains: https://eslint.org/docs/rules/no-use-before-define
Second is that after this usage of this hook there is duplicated code in 3 projects, maybe at this point it should be exported from runtime ( but sincerely I don't think runtime is the project with that responsibility and would recommend a utils project, I don't know if this pattern is used in vtex

Reply:
I totally agree on creating a separate app for exporting this function, on it:
https://github.com/vtex-apps/on-view/pull/1

Reply:
Nice! On the other hand, can you continue with the review before that PR will be merged? I client depends on this for go live. I will make another PR after that that one will be merged in order to import the function

Reply:
meanwhile I updated the project to use exported function from on-view. Can you take another look on it?